src/hotspot/share/gc/z/zOopClosures.hpp

```diff
@@ -67,14 +67,22 @@ class ZPhantomIsAliveObjectClosure : public BoolObjectClosure {
 
 class ZPhantomKeepAliveOopClosure : public ZRootsIteratorClosure {
 public:
-  virtual void do_oop(oop* p);
-  virtual void do_oop(narrowOop* p);
+  virtual void do_oop(oop* p) override;
+  virtual void do_oop(narrowOop* p) override;
+
+  virtual ZNMethodEntry nmethod_entry() const override {
+    return ZNMethodEntry::VerifyDisarmed;
+  }
 };
 
 class ZPhantomCleanOopClosure : public ZRootsIteratorClosure {
 public:
-  virtual void do_oop(oop* p);
-  virtual void do_oop(narrowOop* p);
+  virtual void do_oop(oop* p) override;
+  virtual void do_oop(narrowOop* p) override;
+
+  virtual ZNMethodEntry nmethod_entry() const override {
+    return ZNMethodEntry::VerifyDisarmed;
```

Comment:
Same here, except the ZPhantomCleanOopClosure is used from ZProcessConcurrentWeakRootsTask which also does not visit any nmethods. Similar argument that if it did, it still would not make sense to verify that nmethods have been disarmed.